src/pybind/mgr/snap_schedule/module.py

```diff
@@ -65,7 +65,9 @@ def snap_schedule_get(self,
         '''
         List current snapshot schedules
         '''
-        use_fs = fs if fs else self.default_fs
+        if not fs:
+            return -errno.EINVAL, '', "filesystem argument is required"
+        use_fs = fs
```

Comment:
do we really need `use_fs`? now since `fs` is mandatory, can't we directly use it? btw `fs: Optional[str] = None` need to changed too

Reply:
>I've left use_fs in place to keep changes minimal

there is no use of `use_fs` now, this variable can be avoided, it is just an extra line now. It would require only a few more changes but i it would look much clean and readable

>I'm not sure about changing fs: Optional[str] = None to fs: str since we may also need to change the position of the mandatory argument in the function prototype; Not sure how python and/or MgrModule handles the delegation to python in such cases

`Optional` type annotation is just for the type checker right, just for hints i guess, i don't think this should cause any trouble. It's same as mentioning `path: str` or just `path`

Reply:
> > I've left use_fs in place to keep changes minimal
> 
> there is no use of `use_fs` now, this variable can be avoided, it is just an extra line now. It would require only a few more changes but i it would look much clean and readable

This has been taken care of.


> 
> > I'm not sure about changing fs: Optional[str] = None to fs: str since we may also need to change the position of the mandatory argument in the function prototype; Not sure how python and/or MgrModule handles the delegation to python in such cases
> 
> `Optional` type annotation is just for the type checker right, just for hints i guess, i don't think this should cause any trouble. It's same as mentioning `path: str` or just `path`

The issue is not about "what type hint is needed here?"
The issue is about making the `fs` argument mandatory and without a default value.
This changes the logic of argument processing completely and imposes a fixed order on the presence of the mandatory arguments. Change in the argument passing order will also affect user experience.

One might argue that printing an error message for making the `fs` argument mandatory will also affect user experience when the docs and type hint code says that the `fs` argument is optional.

If we are okay with these changes, then I propose to make the `fs` argument as the very first argument for every snap-schedule command. We can update the docs accordingly.

@vshankar @gregsfortytwo Does my proposal sound good ?

Reply:
oh, and these changes will also need to update relevant YAMLs and python tests

Reply:
or do we just make the `fs` argument mandatory with a default value of `None` and then complain in the code that the argument can't be empty ?
but python argument processing doesn't like this because there would be other `Optional` arguments preceding the mandatory `fs` argument in the function argument list